alembic/versions/2023_07_18_1727-2e42e5319395_md5sum_fileondisk.py

```diff
@@ -19,13 +20,13 @@
 def upgrade() -> None:
     # ### commands auto generated by Alembic - please adjust! ###
     op.add_column('cutouts', sa.Column('md5sum', sa.UUID(), nullable=True))
-    op.add_column('cutouts', sa.Column('md5sum_extensions', sa.ARRAY(sa.UUID()), nullable=True))
+    op.add_column('cutouts', sa.Column('md5sum_extensions', postgresql.JSONB(), nullable=True))
```

Comment:
yeah, but how else are you going to store it as a dictionary?

Reply:
You aren't.   The other way is an ordered array as I had it before.  I think the inefficiency is probably OK for the images table, but at some point we are going to want to start paying attention to all the times we add to the length of table rows thinking that it doesn't matter.

Reply:
What happens if I save one out of two extensions, and not in the order they are listed in `filepath_extensions`?

Reply:
I'd have to go look at the code, but what I had before looked through the filepath_extensions to find the right index.

I think the way to do it is to elimiate the filepath_extensions field, and have md5sum_extensions be a JSONB dictionary, and the keys of the dictionary say which extensions exist.  They are None if the md5sum isn't calculated (not saved to the archive, or, if archive is null, not saved to local disk).  That amount of storage is very similar to having two ordered arrays, I think, and it has the advantage of not duplicating information, which removes a way of introducing inconsistency.

Reply:
I think that with the two arrays you may get in trouble if you do the order wrong, but maybe I missed something. 
In any case I think keeping them all in one dictionary is good in terms of saving space and consistency, I just worry that it is not entirely obvious where to find things. When you are going to read an extension, looking for the key of the md5sum_extensions is not the most intuitive place... maybe simply renaming that single dictionary something more informative might do the trick, I will think about it. 

Honestly I really wish we could have gotten rid of all these extensions, it would simplify everything. I assume we are stuck with them because of some tools that expect separate FITS files?

Reply:
We could get rid of extensions.  We'd just have to write other temporary files later when using external tools, but we're going to have to do some of that anyway.

I predict, though, that if we did, we'd curse ourselves later, as we discovered that some of the things we wanted to save along side images didn't naturally fit into (say) FITS format, or whatever.